lib/src/data/communication/api/request_factory.dart

```diff
@@ -0,0 +1,59 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:allthenews/src/data/communication/api/api_errors.dart';
+import 'package:allthenews/src/data/communication/api/api_key_local_repository.dart';
+import 'package:allthenews/src/domain/authorization/api_key.dart';
+import 'package:async/async.dart';
+import 'package:http/http.dart' as http;
+
+abstract class _Constants {
+  static const Duration timeoutDuration = Duration(seconds: 15);
+}
+
+class RequestFactory {
+  ApiKeyLocalRepository _apiKeyLocalRepository = ApiKeyLocalRepository();
+  ApiKey _apiKey;
+  final String baseUrl;
+
+  RequestFactory(this.baseUrl) {
+    _getApiKey();
```

Comment:
asynchroniczny call w konstruktorze?

Reply:
metoda do pobrania klucza jest asynchroniczna, widzę tu dwa wyjścia:
- albo zostawiamy jak jest, ale trzeba jeszcze dorobić metodę, która przed użyciem obiektu klasy nam powie, że inicjalizacja obiektu została zakończona,
- albo wywoływać pobieranie klucza na stworzonym obiekcie klasy,
co myślisz? masz lepszy pomysł?

Reply:
pobranie klucza api jest asynchroniczne, poprawię

Reply:
ogólnie wydaje mi się, że średnio żeby konstruktor wywoływał coś asynchronicznie - może lepiej, żeby przy pierwszym requeście pobrał sobie ten klucz jak go nie ma, a później korzystał z zacacheowanego?

Reply:
a co powiesz na fabrykę, która wywoła prywatny konstruktor i żeby uzyskać obiekt będzie trzeba zrobić await?

Reply:
nie wiem czy dobrze zrozumiałem, ale w takim wypadku jeżeli dostarczysz to do drzewa dependencji do get_it to przy inicjalizacji DI będzie trzeba wywołać metodę fabrykującą dodając await? Jeżeli tak, to imo słabe :P

Reply:
tak by właśnie było, pobranie klucza przy pierwszym requeście wydaje się najlepszą opcją